akka-docs-dev/rst/scala/code/docs/stream/cookbook/RecipeKeepAlive.scala

```diff
@@ -10,68 +10,16 @@ class RecipeKeepAlive extends RecipeSpec {
   "Recipe for injecting keepalive messages" must {
 
     "work" in {
-
-      type Tick = Unit
-
-      val tickPub = TestPublisher.probe[Tick]()
-      val dataPub = TestPublisher.probe[ByteString]()
-      val sub = TestSubscriber.manualProbe[ByteString]()
-      val ticks = Source(tickPub)
-
-      val dataStream = Source(dataPub)
       val keepaliveMessage = ByteString(11)
-      val sink = Sink(sub)
 
       //#inject-keepalive
-      val tickToKeepAlivePacket: Flow[Tick, ByteString, Unit] = Flow[Tick]
-        .conflate(seed = (tick) => keepaliveMessage)((msg, newTick) => msg)
-
-      val graph = RunnableGraph.fromGraph(FlowGraph.create() { implicit builder =>
-        import FlowGraph.Implicits._
-        val unfairMerge = builder.add(MergePreferred[ByteString](1))
-
-        // If data is available then no keepalive is injected
-        dataStream ~> unfairMerge.preferred
-        ticks ~> tickToKeepAlivePacket ~> unfairMerge ~> sink
-        ClosedShape
-      })
+      import scala.concurrent.duration._
+      val injectKeepAlive: Flow[ByteString, ByteString, Unit] =
+        Flow[ByteString].injectIfIdle(1.second, () => keepaliveMessage)
```

Comment:
In order to keep permutations low, how about we repurpose `idleTimeout` to take a function () => B and then someone can have it be () => throw TimeoutException if they want?